types/underscore/index.d.ts

```diff
@@ -813,54 +828,38 @@ declare module _ {
-        * of which contains all of the first elements in the input arrays, the second of which
-        * contains all of the second elements, and so on. Use with apply to pass in an array
-        * of arrays
-        * @param arrays The arrays to unzip.
-        * @return Unzipped version of `arrays`.
-        **/
-        unzip(...arrays: any[][]): any[][];
-
-        /**
-        * Converts arrays into objects. Pass either a single list of [key, value] pairs, or a
-        * list of keys, and a list of values.
-        * @param keys Key array.
-        * @param values Value array.
-        * @return An object containing the `keys` as properties and `values` as the property values.
-        **/
-        object<TResult extends {}>(
-            keys: _.List<string>,
-            values: _.List<any>): TResult;
+         * Merges together the values of each of the `lists` with the values at
+         * the corresponding position. Useful when you have separate data
+         * sources that are coordinated through matching list indexes.
+         * @param lists The lists to zip.
+         * @returns The zipped version of `lists`.
+         **/
+        zip(...lists: List<any>[]): any[][];
 
         /**
-        * Converts arrays into objects. Pass either a single list of [key, value] pairs, or a
-        * list of keys, and a list of values.
-        * @param keyValuePairs Array of [key, value] pairs.
-        * @return An object containing the `keys` as properties and `values` as the property values.
-        **/
-        object<TResult extends {}>(...keyValuePairs: any[][]): TResult;
+         * The opposite of zip. Given a list of lists, returns a series of new
+         * arrays, the first of which contains all of the first elements in the
+         * input lists, the second of which contains all of the second
+         * elements, and so on.
+         * @param lists The lists to unzip.
+         * @returns The unzipped version of `lists`.
+         **/
+        unzip(lists: List<List<any>>): any[][];
 
         /**
-        * @see _.object
-        **/
-        object<TResult extends {}>(
-            list: _.List<any>,
-            values?: any): TResult;
+         * Converts lists into objects. Pass either a single `list` of
+         * [key, value] pairs, or a `list` of keys and a list of `values`.
+         * Passing by pairs is the reverse of pairs. If duplicate keys exist,
+         * the last value wins.
+         * @param list A list of [key, value] pairs or a list of keys.
+         * @param values If `list` is a list of keys, a list of values
+         * corresponding to those keys.
+         * @retursn An object comprised of the provided keys and values.
+         **/
+        object<V extends List<any>, TValueList extends List<any> | undefined = undefined>(
+            list: V,
+            values?: TValueList
+        ): Dictionary<_ObjectValue<TypeOfList<V>, TValueList>>;
```

Comment:
How do you feel about type parameter names that are longer than a single letter @jgonggrijp? I'm a little tempted as part of cleanup to maybe make some updates to the type parameter names in this definition to help better document what they are and how they differ between interfaces (e.g. `V`'s in `UnderscoreStatic` would become `TCollection` or `TList`, `V`'s and `T`'s in `Underscore` and `_Chain` would become `TWrapped` and `TCollectionItem` respectively, etc.

Reply:
I like single letters when it is easy to deduce from the context what the letter stands for. I also like `T` when the context doesn't really care what kind of type it is. Otherwise, I totally agree that descriptive names are better. Specifically, I agree that `V` is unclear. I'm still not sure what that letter is supposed to stand for; the type that I'd consider the "value type" in context is often something other than `V`.